v19.1/build-a-python-app-with-cockroachdb-sqlalchemy.md

```diff
@@ -22,6 +22,12 @@ We have tested the [psycopg2 driver](http://initd.org/psycopg/docs/) and [SQLAlc
 The example code on this page uses Python 3.
 {{site.data.alerts.end}}
 
+{{site.data.alerts.callout_danger}}
+SQLAlchemy relies on the existence of foreign keys to generate [`JOIN` expressions](joins.html) from your application code. If you remove foreign keys from your schema, SQLAlchemy won't generate joins for you. Instead, you'll have to issue raw SQL (via [`engine.execute()`](https://docs.sqlalchemy.org/en/13/core/connections.html?highlight=engine.execute#sqlalchemy.engine.Engine.execute)) or do the equivalent work in your application.
```

Comment:
Instead of falling back to raw SQL, you can also use a "custom foreign condition" by adding a `relationship` field to your table objects.

https://stackoverflow.com/questions/37806625/sqlalchemy-create-relations-but-without-foreign-key-constraint-in-db

https://docs.sqlalchemy.org/en/13/orm/join_conditions.html#creating-custom-foreign-conditions

(the sqlalchemy docs are rather opaque about this so the SO answer is probably worth linking to instead)

Reply:
I've updated this section to point users to the custom foreign condition SO link and removed the (potentially confusing) link to the SQLAlchemy docs, since it might not be necessary to solve the user's problem.  PTAL.

Reply:
Thanks @bdarnell -- I tried following the docs and this still wasn't working (as you noticed, the docs weren't that clear). I gave up after that and just did the joins at the app level. Good to know there is a more comprehensive SO solution.